version1/04 - Web services.md

```diff
@@ -468,6 +468,128 @@ Content-Type: */*
+- An **index document** is a set of fields and their corresponding values stored in a search index.
+- Each index document has a **doctype** which corresponds to a set of Noark 5 fields stored in the index document.
+- The supported doctypes and the object types (or fields) included in them are:
+
+| doctype            | Object types and fields                                                                                                                                                                                                                                                                                                |
+|:-------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Tekst              | Dokument, Tekst (of the last Dokumentversjon), AbstraktRegistrering, Noekkelord, Korrespondansepart, virksomhetsspesifikkeMetadata, Klasse (secondary of the AbstraktRegistrering and the AbstraktMappe and primary of the AbstraktMappe), Klassifikasjonssystem (secondary), AbstraktMappe, Sakspart, Arkivdel, Arkiv |
+| Korrespondansepart | Korrespondansepart                                                                                                                                                                                                                                                                                                     |
+| Registrering       | AbstraktRegistrering (all types), Noekkelord, Korrespondansepart, virksomhetsspesifikkeMetadata, Klasse (secondary), Klassifikasjonssystem (secondary), AbstraktMappe, Arkivdel, Arkiv                                                                                                                                 |
+| Mappe              | AbstraktMappe (all types), Noekkelord, Sakspart, virksomhetsspesifikkeMetadata                                                                                                                                                                                                                                         |
+| Arkivdel           | Arkivdel                                                                                                                                                                                                                                                                                                               |
+
+## **search**
+
+Performs free text search in index documents.
+
+**Request**
+
+``` text
+POST /rms/api/public/noark5/v1/full-text/search HTTP/1.1
+Authorization: Bearer ACCESS_TOKEN
+Content-Type: application/json
+
+{
+  "doctype": string,
+  "query": string,
+  "offset: number,
+  "limit": number
+}
+```
+
+###### Details
```

Comment:
I would also add a comment to `query` that other search fields may be returned too and clients must not make an assumption about the exact set of fields returned by this endpoint.

Reply:
Added.